### DISCUSSION TITLE
Improvements to Sidekiq::TransactionAwareClient

### DISCUSSION THREAD

**Discussion by ngan** [sidekiq/sidekiq#5672] [Ideas]
I think `Sidekiq::TransactionAwareClient` is great. At Gusto, we've leaned into `Sidekiq.transactional_push!` heavily but would love to see some improvements. `TransactionAwareClient` does not support `push_bulk`, understandably. However, it would be nice to be able to actually support `push_bulk` but perhaps have limits on the number of jobs in memory, and erring if that limit is reached. And/Or, allow an option to be set on `TransactionAwareClient` that raises an error when `push_bulk` is called while `AfterCommitEverywhere.in_transaction?` is `true`.

Currently we're using the [isolator gem](https://github.com/palkan/isolator) to catch job enqueueing while in a DB transaction, but it would be nice to drop it completely if this detection was baked into `TransactionAwareClient`.

**Comment by mperham** [comment]
I think the best approach here would be to open a PR with a proposed improvement so we can talk about actual implementation details and tradeoffs. No promises but that's more grounded than a high-level discussion. WDYT?
